### PR TITLE
[Enhancement] Support abort lake table transaction

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 87 (31.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|1|00.00%|[188]|
|:large_blue_circle:|com/starrocks/lake/Utility.java|0|19|00.00%|[17, 21, 22, 24, 25, 26, 28, 34, 35, 36, 37, 38, 39, 40, 42, 43, 44, 45, 46]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|26|00.00%|[34, 110, 111, 112, 115, 118, 119, 120, 121, 123, 126, 127, 128, 130, 132, 133, 134, 135, 137, 138, 141, 142, 143, 144, 145, 146]|
|:large_blue_circle:|com/starrocks/common/NoAliveBackendException.java|0|4|00.00%|[7, 8, 11, 12]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|[277]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|13|20|65.00%|[211, 223, 224, 225, 226, 227, 231]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|14|16|87.50%|[1260, 1267]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 90 (30.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|1|00.00%|[188]|
|:large_blue_circle:|com/starrocks/lake/Utility.java|0|20|00.00%|[20, 28, 29, 30, 32, 33, 34, 36, 42, 43, 44, 45, 46, 47, 48, 50, 51, 52, 53, 54]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|28|00.00%|[34, 109, 110, 111, 112, 115, 118, 119, 120, 122, 125, 126, 129, 130, 131, 133, 135, 136, 137, 138, 140, 141, 144, 145, 146, 147, 148, 149]|
|:large_blue_circle:|com/starrocks/common/NoAliveBackendException.java|0|4|00.00%|[7, 8, 11, 12]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|[277]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|13|20|65.00%|[211, 223, 224, 225, 226, 227, 231]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|14|16|87.50%|[1260, 1267]|

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 90 (30.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/compaction/CompactionDispatchDaemon.java|0|1|00.00%|[188]|
|:large_blue_circle:|com/starrocks/lake/Utility.java|0|20|00.00%|[20, 28, 29, 30, 32, 33, 34, 36, 42, 43, 44, 45, 46, 47, 48, 50, 51, 52, 53, 54]|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|28|00.00%|[34, 109, 110, 111, 112, 115, 118, 119, 120, 122, 125, 126, 129, 130, 131, 133, 135, 136, 137, 138, 140, 141, 144, 145, 146, 147, 148, 149]|
|:large_blue_circle:|com/starrocks/common/NoAliveBackendException.java|0|4|00.00%|[7, 8, 11, 12]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|1|00.00%|[277]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|13|20|65.00%|[211, 223, 224, 225, 226, 227, 231]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|14|16|87.50%|[1260, 1267]|

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/lake/Utility.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.starrocks.common.UserException;
+import com.starrocks.server.GlobalStateMgr;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Utility {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/Utility.java:19]:
```suggestion
public class Utils {
```

File: fe/fe-core/src/main/java/com/starrocks/lake/Utility.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.lake;
+
+import com.clearspring.analytics.util.Lists;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/Utility.java:5]:
```suggestion
import com.google.common.collect.Lists;
```

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 3]
```diff
@@ -1299,42 +1305,9 @@ private boolean unprotectAbortTransaction(long transactionId, boolean abortPrepa
         transactionState.setReason(reason);
         transactionState.setTransactionStatus(TransactionStatus.ABORTED);
         unprotectUpsertTransactionState(transactionState, false);
-        for (PublishVersionTask task : transactionState.getPublishVersionTasks().values()) {
-            if (task != null) {
-                AgentTaskQueue.removeTask(task.getBackendId(), TTaskType.PUBLISH_VERSION, task.getSignature());
-            }
-        }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1306]: Why remove?

File: fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java [hunk 4]
```diff
@@ -2,22 +2,36 @@
 
 package com.starrocks.transaction;
 
+import com.clearspring.analytics.util.Lists;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java:5]:
```suggestion
import com.google.common.collect.Lists;
```

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java (hunk 3 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1306]: `transactionState.getPublishVersionTasks()` would always be empty here.

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1306]: how to guarantee always be empty?

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1306]: Only transactions of committed state will create publish version tasks but the transaction state must not be committed here, as you can see from the above lines of codes.

**Review by wyb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_admit_test

**Review by tracymacding — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**